HACKING.adoc

```diff
@@ -519,6 +519,47 @@ third-party code, by installing a local <<opam-switch,opam switch>>: opam
 packages tend to be compatible with released versions of the compiler, whereas
 most packages are incompatible with the in-progress development version.
 
+
+=== License
+
+The OCaml compiler is licensed under the LGPL version 2.1, with
+a linking exception. See link:LICENSE[] for the full license
+text. Using the https://spdx.org/licenses/[SPDX] standard for license
+identifiers, this is the license "LGPL-2.1 WITH
+https://spdx.org/licenses/OCaml-LGPL-linking-exception.html[OCaml-LGPL-linking-exception]".
+Let us include here a few explanations about the intent of this
+licensing choice -- note that only the license text has legal value.
+
+The broad intent of the LGPL is that if you modify the OCaml compiler
+or standard library codebase, and distribute the result of the
+modification to your users, then you have to provide the source code
+of your version -- the LGPL, like the GPL and unlike the BSD and MIT
+licenses, is "hereditary", modified versions of LGPL programs must
+remain free software under the same license. (For more general
+discussion of the LGPL license, see the
+https://www.gnu.org/licenses/gpl-faq.en.html[FAQ] maintained by the
+GNU project.)
+
+On the other hand, if your code is not derived from the OCaml standard
+library or compiler codebase, it only uses it, then you can use the
+licensing terms of your choice. We use an explicit "linking exception"
+precisely to make it clear that either static or dynamic linking with
+the standard library or other compiler libraries is fine. Linking
+exceptions are a common practice for language compilers and runtime
+libraries, for example the GCC project uses such an exception, see the
+https://en.wikipedia.org/wiki/GPL_linking_exception[Wikipedia article]
+on linking exceptions for more details.
+
+Finally: If you have your own fork of the OCaml compiler and standard
+library, and distribute it to your own users under the same license
+that we use, then your users will also benefit from the linking
+exception.
+
+OCaml has been used extensively in both free software and proprietary
+software projects, including in large public institutions and private
+companies where its licensing terms were reviewed by lawyers. We are
+not aware of any particular issue with its licensing terms.
```

Comment:
Yet the Caml Consortium provides its members with a less restrictive license, something that some Consortium members say is necessary for their business.  What about removing this paragraph altogether?  It sounds strangely defensive anyway.

Reply:
Done.